drivers/led_strip/apa102.c

```diff
@@ -93,6 +96,23 @@ static int apa102_init(struct device *dev)
 	data->cfg.operation =
 		SPI_OP_MODE_MASTER | SPI_TRANSFER_MSB | SPI_WORD_SET(8);
 
+#if DT_INST_SPI_DEV_HAS_CS_GPIOS(0)
+	data->cs_ctl.gpio_dev =
+		device_get_binding(DT_INST_SPI_DEV_CS_GPIOS_LABEL(0));
+	if (!data->cs_ctl.gpio_dev) {
+		return -ENODEV;
+	}
+	data->cs_ctl.gpio_pin = DT_INST_SPI_DEV_CS_GPIOS_PIN(0);
+	data->cs_ctl.delay = 0;
+
+	data->cfg.cs = &data->cs_ctl;
+
+	int flags = DT_INST_SPI_DEV_CS_GPIOS_FLAGS(0);
```

Comment:
Although it's rare, not all GPIO specifiers support a flags field. Can you please add the appropriate ifdeffery to support this case? See samples/basic/blinky/src/main.c and samples/basic/button/src/main.c for two different ways to achieve this.

Reply:
At the time of writing, there isn't a clear way to check if the flags field exists without making the code some-what unreadable. Instead #25827 would be a good way to address this